### PR TITLE
[EOBS-812] Allocating a doctor deallocates them from other wards

### PR DESCRIPTION
**Pull request by moderatemisbehaviour**
When doctors are allocated to a new ward using the *Medical Shift Change* page they are no longer deallocated from their other wards.

---

Before this pull request can be merged the following must be true.
- [x] Unit tests pass (Travis integration).
- [x] No code quality issues (Codacy integration).
- [x] Approval from at least one developer and at least one tester.

If you are a BJSS contributor there are some additional conditions.
- [ ] All client module unit tests pass.
- [x] The *Pull Request* field in the work tab of the JIRA issue is populated.
- [ ] The JIRA issue contains a description of the root cause and the solution in the comments.
- [x] There are no bugs against the JIRA issues in the current sprint.

### CODE REVIEW

File: nh_clinical/tests/common/nh_clinical_test_utils.py [hunk 1]
```diff
@@ -233,18 +233,21 @@ def create_hca(self, location_id=None):
...
+        if not location_ids:
+            location_ids = [self.ward.id, self.bed.id]
         # Create doctor and associate them with bed location and doctor role.
         self.doctor = self.user_model.create({
             'name': 'Doctor Acula',
             'login': 'Dr_Acula',
             'password': 'Dr_Acula',
             'category_id': [[4, self.doctor_role.id]],
-            'location_ids': [[6, 0, [self.ward.id, self.bed.id]]]
+            'location_ids': [[6, 0, location_ids]]
```
Comment (Gimpneek) [inline nh_clinical/tests/common/nh_clinical_test_utils.py]: Might want to ensure that `location_ids` be a list cos otherwise will blow up

File: nh_clinical/tests/nh_clinical_doctor_allocation/test_submit_users.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            })
+        doctor_allocation_wizard.submit_users()
+
+        self.assertIn(ward_1, self.doctor.location_ids)
+        self.assertIn(ward_2, self.doctor.location_ids)
+
+    def test_new_ward_allocated_when_already_allocated_to_other_wards(self):
+        ward_3 = self.test_utils.create_location(
+            'ward', self.test_utils.hospital.id)
+        ward_3_shift_coordinator = self.test_utils.create_shift_coordinator(
```
Comment (Gimpneek) [inline nh_clinical/tests/nh_clinical_doctor_allocation/test_submit_users.py]: There's some duplication on setting up ward 3 shift coordinator. Should this move to `setUp`?

File: nh_clinical/wizard/user_allocation.py [hunk 3]
```diff
@@ -493,16 +493,29 @@ def submit_users(self, cr, uid, ids, context=None):
         if not isinstance(ids, list):
             ids = [ids]
         wiz = self.browse(cr, uid, ids[0], context=context)
-        respallocation_pool = self.pool[
+        reallocation_pool = self.pool[
```
Comment (Gimpneek) [inline nh_clinical/wizard/user_allocation.py]: Why was this renamed? The Responsibility allocation model is not the same as the reallocation model.

File: nh_clinical/tests/common/nh_clinical_test_utils.py (hunk 1 above)
Comment (moderatemisbehaviour) [reply nh_clinical/tests/common/nh_clinical_test_utils.py]: Good point.

File: nh_clinical/tests/nh_clinical_doctor_allocation/test_submit_users.py (hunk 2 above)
Comment (moderatemisbehaviour) [reply nh_clinical/tests/nh_clinical_doctor_allocation/test_submit_users.py]: Yep thanks.

File: nh_clinical/wizard/user_allocation.py (hunk 3 above)
Comment (moderatemisbehaviour) [reply nh_clinical/wizard/user_allocation.py]: Looked like a typo. Will rename.

**Comment by Nayira** [comment]
Test for EOBS-812 pass. :shipit:
